demo/lightning/checkpoint/multinode/strategies.py

```diff
@@ -120,6 +123,54 @@ def load_checkpoint(self, checkpoint_path):
         return metadata
 
 
+class AsyncDatafluxFSDPStrategy(DatafluxFSDPStrategy):
```

Comment:
Recording what we discussed here:

- Consider implementing the teardown method
- See if it's possible to put this logic directly in `DatafluxFSDPStrategy` with just a boolean async flag or something

Reply:
Great suggestions. I'm working on those improvements and will update this PR with those changes.